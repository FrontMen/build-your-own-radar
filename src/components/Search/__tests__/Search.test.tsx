import React from 'react';
import { Search } from '..';
import {
  withAllProviders,
  AllProvidersWrapper,
  getSelector,
} from 'test/helpers';
import { mount } from 'enzyme';

const setHighlighted = jest.fn();

const inputSelector = getSelector('search-input');
const dropdownSelector = getSelector('search-content');

describe('Search', () => {
  /* test to check that the mobile and larger snapshot don't change unintentionally */
  it('render search icon and input field', () => {
    const { container, getByTestId } = withAllProviders(
      <Search setHighlighted={setHighlighted} />,
    );

    expect(getByTestId('search-icon')).toBeTruthy();
    expect(getByTestId('search-input')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('should render dropdown if input has a matching value', () => {
    const wrapper = mount(<Search setHighlighted={setHighlighted} />, {
      wrappingComponent: AllProvidersWrapper,
    });

    //there should not be dropdown when input is empty
    expect(wrapper.find(dropdownSelector).at(0)).toHaveLength(0);

    wrapper
      .find(inputSelector)
      .at(0)
      .simulate('change', {
        target: {
          value: 'a',
        },
      });

    let dropdown = wrapper.find(dropdownSelector).at(0);

    expect(dropdown).toHaveLength(1);
    expect(dropdown.at(0).find(getSelector('search-ringName')).length).toBeGreaterThan(0);
    expect(dropdown.at(0).find(getSelector('search-technology')).length).toBeGreaterThan(0);

    wrapper
      .find(inputSelector)
      .at(0)
      .simulate('change', {
        target: {
          value: 'there is no matching technology with such name',
        },
      });

    //there should not be dropdown when there are no matching technologies
    dropdown = wrapper.find(dropdownSelector).at(0);

    expect(dropdown).toHaveLength(0);

  });
});
