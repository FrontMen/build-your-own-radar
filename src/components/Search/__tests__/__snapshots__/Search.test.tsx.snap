// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render search icon and input field 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  position: relative;
  padding: 8px 0;
}

.c1 {
  -webkit-flex: 0 0 1.5em;
  -ms-flex: 0 0 1.5em;
  flex: 0 0 1.5em;
  height: 2em;
  width: 2em;
  margin: 5px;
}

.c3 {
  border: none;
  border-bottom: 2px solid;
  border-color: #767676;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  outline: none;
  -webkit-appearance: none;
  background: none;
}

.c3:focus {
  border-color: #526de4;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

<div>
  <div
    class="c0"
    data-testid="subnav-search-container"
  >
    <svg
      class="c1"
      data-testid="search-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M443.5 420.2L336.7 312.4c20.9-26.2 33.5-59.4 33.5-95.5 0-84.5-68.5-153-153.1-153S64 132.5 64 217s68.5 153 153.1 153c36.6 0 70.1-12.8 96.5-34.2l106.1 107.1c3.2 3.4 7.6 5.1 11.9 5.1 4.1 0 8.2-1.5 11.3-4.5 6.6-6.3 6.8-16.7.6-23.3zm-226.4-83.1c-32.1 0-62.3-12.5-85-35.2-22.7-22.7-35.2-52.9-35.2-84.9 0-32.1 12.5-62.3 35.2-84.9 22.7-22.7 52.9-35.2 85-35.2s62.3 12.5 85 35.2c22.7 22.7 35.2 52.9 35.2 84.9 0 32.1-12.5 62.3-35.2 84.9-22.7 22.7-52.9 35.2-85 35.2z"
      />
    </svg>
    <div
      class="c2"
    >
      <input
        class="c3"
        data-testid="search-input"
        placeholder="words.search"
        type="search"
        value=""
      />
    </div>
  </div>
</div>
`;
